### ISSUE TITLE
Java "ClassNotFoundException" due to unzipped code passed to lambci/lambda

### ISSUE THREAD

**Issue by sjmatta** [localstack/localstack#1979]
`awslocal lambda create-function --function-name helloworld --zip-file fileb://target/hello-world-0.1.jar --runtime java11 --handler hello.world.HelloWorldFunction::get --role whatever`

`awslocal lambda invoke --function-name helloworld out.txt && cat out.txt && rm out.txt`

This fails due to `Class not found: hello.world.HelloWorldFunction: java.lang.ClassNotFoundException`

The same function works fine when deploying to AWS with the same commands.

I dug in a bit, and I see that `/tmp/localstack/zipfile.e89d9c5d` has one file, `original_lambda_archive.zip`. When I unzip this and re-invoke the Lambda, it's successful.

According to https://github.com/lambci/docker-lambda, it's expecting unzipped.

**Comment by sjmatta** [comment]
If I'm in `/tmp/localstack/zipfile.e89d9c5d` and I run:
`docker run --rm -v "$PWD":/var/task lambci/lambda:java11 hello.world.HelloWorldFunction::get`
Then I get the same error. If I unzip the zip file, then rerun, it's successful.

**Comment by sjmatta** [comment]
Issue may be here: https://github.com/localstack/localstack/blob/03c6f4bf250edd18eb818ed65090f508636b0bff/localstack/services/awslambda/lambda_api.py#L593

Which is at odds with https://github.com/lambci/docker-lambda

I'm going to try adding a `/lib` folder as a workaround for now, but I think all Java JARs should be unzipped.

**Comment by sjmatta** [comment]
I may be able to open a PR for this soon, but in the meantime, anyone who encounters this: there's a temporary workaround of adding `/src/main/resources/lib/empty.jar` to your Java project.

**Comment by whummer** [comment]
Thanks for reporting, and thanks for opening the pull request @sjmatta . Any chance you can share the your `hello-world-0.1.jar` JAR file? I just did a quick test, and it seems that uploading a JAR file directly in fact works:
```
$ aws lambda create-function --function-name test-java --runtime java8 --role r1  --handler cloud.localstack.sample.LambdaHandler --zip-file fileb://localstack/ext/java/target/localstack-utils-0.2.0-tests.jar
$ awslocal lambda invoke --function-name test-java /tmp/foobar
{
    "StatusCode": 200
}
```

We also have an integration test for direct upload of JAR files here: https://github.com/localstack/localstack/blob/77d4061526741d9063d388de0a183347b5d9c428/tests/integration/test_lambda.py#L520-L529

My suspicion would be that the zip file is actually the JAR file with a different file ending, in your case something like the following (please note that LocalStack should be able to add this ZIP file to the Java classpath when running the Lambda):
```
$ tar -tf /tmp/localstack/zipfile.e89d9c5d/original_lambda_archive.zip
```

The main reason we're trying to run Lambdas from JAR files directly without extracting is for performance reasons. We've had cases of 50+ MB JAR files with with thousands of classes, which can take in the order of up to half a minute to exract on some systems. Hence, my goal here would be to understand the root cause for this particular use case before we proceed with the PR in #1982. Thanks

**Comment by whummer** [comment]
Please ignore my previous comment @sjmatta - turns out that it works with Java 8 but not with Java 11, due to the issue described here: #1985 . This should be fixed in #1987. Please give it a try with the latest version once the PR is merged and the image has been built in Travis-CI. Thanks
